rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -41,110 +41,144 @@ typedef struct BoundingBoxCord_
+    std::string _img_name = "";
+    ImgSize _img_size = {};
+} MetaDataInfo;
 
 struct MetaData
 {
-    int& get_label() { return _label_id; }
-    BoundingBoxCords& get_bb_cords() { return _bb_cords; }
-    BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { return _bb_cords_xcycwh; }
-    BoundingBoxLabels& get_bb_labels() { return _bb_label_ids; }
-    void set_bb_labels(BoundingBoxLabels bb_label_ids)
-    {
-        _bb_label_ids = std::move(bb_label_ids);
-        _object_count = _bb_label_ids.size();
-    }
-    ImgSize& get_img_size() {return _img_size; }
-    int get_object_count() { return _object_count; }
-    std::vector<size_t> get_bb_label_dims()
-    {
-        _bb_labels_dims = {_bb_label_ids.size()};
-        return _bb_labels_dims;
-    }
-    std::vector<size_t> get_bb_cords_dims()
-    {
-        _bb_coords_dims = {_bb_cords.size(), 4};
-        return _bb_coords_dims;
-    }
-protected:
-    BoundingBoxCords _bb_cords = {}; // For bb use
-    BoundingBoxCords_xcycwh _bb_cords_xcycwh = {}; // For bb use
-    BoundingBoxLabels _bb_label_ids = {};// For bb use
-    ImgSize _img_size = {};
-    int _label_id = -1; // For label use only
-    std::vector<size_t> _bb_labels_dims = {};
-    std::vector<size_t> _bb_coords_dims = {};
-    int _object_count = 0;
+    virtual std::vector<int>& get_label() { };
+    virtual int get_object_count() { };
+    virtual std::vector<size_t> get_label_dims() { };
+    virtual void set_labels(Labels label_ids) { };
+    virtual BoundingBoxCords& get_bb_cords() { };
+    virtual BoundingBoxCords_xcycwh& get_bb_cords_xcycwh() { };
+    virtual std::vector<size_t> get_bb_cords_dims() { };
+    virtual void set_bb_cords_xcycwh(BoundingBoxCords_xcycwh bb_cords_xcycwh) { };
+    virtual void set_bb_cords(BoundingBoxCords bb_cords) { };
+    ImgSize& get_img_size() {return _info._img_size; }
+    std::string& get_image_name() { return _info._img_name; }
+    uint& get_image_id() { return _info._img_id; }
+    void set_img_size(ImgSize img_size) { _info._img_size = std::move(img_size); }
+    void set_img_id(uint img_id) { _info._img_id = std::move(img_id); }
+    void set_img_name(std::string img_name) { _info._img_name = std::move(img_name); }
+    void set_metadata_info(MetaDataInfo info) { _info = std::move(info); }
+    protected:
+    MetaDataInfo _info;
 };
 
 struct Label : public MetaData
 {
     Label(int label)
     {
-        _label_id = label;
+        _label_ids.resize(1);
+        _label_ids[0] = label;
         _object_count = 1;
     }
-    Label(){ _label_id = -1; }
+    Label()
+    {
+        _label_ids.resize(1);
+        _label_ids[0] = -1;
+    }
+    std::vector<int>& get_label() { return _label_ids; }
+    int get_object_count() { return _object_count; }
+    std::vector<size_t> get_label_dims()
+    {
+        _labels_dims = {_label_ids.size()};
+        return _labels_dims;
+    }
+    void set_labels(Labels label_ids)
+    {
+        _label_ids = std::move(label_ids);
```

Comment:
Shouldn't we resize / reserve this variable?

Reply:
I think it not required @shobana-mcw